src/components/TextLink.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {ForwardedRef, forwardRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, StyleProp, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type LinkProps = {
+    /** Link to open in new tab */
+    href: string;
+
+    onPress?: undefined;
+};
+
+type PressProps = {
+    href?: undefined;
+
+    /** Overwrites the default link behavior with a custom callback */
+    onPress: () => void;
+};
+
+type TextLinkProps = (LinkProps | PressProps) & {
+    /** Text content child */
+    children: ReactElement;
+
+    /** Additional style props */
+    style?: StyleProp<TextStyle>;
+
+    /** Callback that is called when mousedown is triggered */
+    onMouseDown?: MouseEventHandler;
+};
+
+function TextLink({href, onPress, children, style, onMouseDown = (event) => event.preventDefault(), ...props}: TextLinkProps, ref: ForwardedRef<RNText>) {
+    const openLink = () => {
+        if (onPress) {
+            onPress();
+        } else {
+            Link.openExternalLink(href);
+        }
+    };
+
+    const openLinkOnTap = (event: GestureResponderEvent) => {
+        event.preventDefault();
+
+        openLink();
+    };
+
+    const openLinkOnEnterKey: KeyboardEventHandler = (event) => {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        event.preventDefault();
+
+        openLink();
+    };
+
+    return (
+        <Text
+            style={[styles.link, style]}
+            role={CONST.ACCESSIBILITY_ROLE.LINK}
+            href={href}
+            onPress={openLinkOnTap}
+            onKeyDown={openLinkOnEnterKey}
+            onMouseDown={onMouseDown}
+            ref={ref}
+            suppressHighlighting
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
```

Comment:
I just noticed that props is an empty object:
![image](https://github.com/Expensify/App/assets/46095609/34464ba3-d1fe-4de4-8873-427d353a1b40)
Most likely we need to extend TextLinkProps with `TextProps` (export it from `src/components/Text.tsx`)

```ts
type TextLinkProps = (LinkProps | PressProps) & TextProps;
```